vizro-core/src/vizro/models/_components/form/dropdown.py

```diff
@@ -44,16 +45,32 @@ def validate_multi(multi, info: ValidationInfo):
     return multi
 
 
-def _add_select_all_option(full_options: OptionsType) -> OptionsType:
+def _add_select_all_option(
+    full_options: OptionsType, component_id: str, value: Optional[Union[SingleValueType, MultiValueType]]
+) -> OptionsType:
     """Adds a 'Select All' option to the list of options."""
-    # TODO: Move option to dictionary conversion within `get_options_and_default` function as here: https://github.com/mckinsey/vizro/pull/961#discussion_r1923356781
-    options_dict = [
-        cast(OptionsDictType, {"label": option, "value": option}) if not isinstance(option, dict) else option
-        for option in full_options
-    ]
-
-    options_dict[0] = {"label": html.Div(["ALL"]), "value": "ALL"}
-    return options_dict
+    checklist_value = (
+        [ALL_OPTION] if value is None or (isinstance(value, list) and len(value) == len(full_options)) else []
+    )
+    full_options = cast(list[OptionsDictType], full_options)
+    all_options = {
+        "label": html.Div(
+            [
+                dcc.Checklist(
+                    options=[{"label": "", "value": ALL_OPTION}],
+                    value=checklist_value,
+                    id=f"{component_id}_checklist_all",
```

Comment:
It's a nitpicking and sorry for that, but let's be consistent with the "rule" where the `id` component property goes on the first place. 
```suggestion
                    id=f"{component_id}_checklist_all",
                    options=[{"label": "", "value": ALL_OPTION}],
                    value=checklist_value,
```